### DISCUSSION TITLE
Creating Signals and fetching data outside of components

### DISCUSSION THREAD

**Discussion by olegKusov** [QwikDev/qwik-evolution#165] [Proposals For Qwik]
### Is your feature request related to a problem?

I have problem with ability to create good architecture of app with qwik. Now Qwik give us React-like experience with hooks, state and lifecycle hooks. And do not give us good way of handling big apps.

### Describe the solution you'd like

Let's look at some state managers. [Reatom](https://www.reatom.dev) give us ability to create Atoms (signals) outside of components and handle data fetching outside of components. 
We can do even this:
![image](https://user-images.githubusercontent.com/28058268/223350008-463c7ddf-b9a0-4a79-b0f4-24d5cd838308.png)

[Effector](https://effector.dev) STM handle logic and data fetching and with Effector we can construct a lot of logic in declarative manner.

![image](https://user-images.githubusercontent.com/28058268/223352318-c5b33f5b-2a9b-4895-9854-7bd914d5b4f7.png)

With [atomic router](https://atomic-router.github.io/) we can even separate data fetching from lifecycle hooks: for example, on page change event Effector with Atomic Router can call Effect with data fetching and call route change via Atomic Router. And know we do not depend on lifecycle hooks - View of App reacts to the model, but not vice versa

Vue give us ability to separate view from model with templates. Angular give templates, DI. And what give us Qwik? How we can handle big apps? A lot of hooks inside components? But Why? Why we can't create new architecture - new way of app building? Why we using the same principles that React gave us many years ago? Yes Qwik is good performant framework but it's not enough today. We need easy way of working with business logic, easy way of scalability. 
Qwik even do not have package for forms - its basic thing for frontend Framework. I wanna have all in one solution if its called 'framework'.

### Describe alternatives you've considered

All in one solution for handling big projects.

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
